### ISSUE TITLE
Tensor with 2-max-values in Max Last stores wrong gradients

### ISSUE THREAD

**Issue by M1ngXU** [coreylowman/dfdx#111]
According to [this](https://github.com/coreylowman/dfdx/blob/d7037549c3df0e0507bbe2269e7aee0f974324ba/src/tensor_ops/impl_max_last.rs#L24) line, if a tensor is something like `[2.0, 2.0, 1.0]`, then both `2.0s` will get a derivative of `1`, though only the first should.
Again, I can’t test this right now, so I am not 100% sure wheather I am right.

**Comment by coreylowman** [comment]
Yeah that's definitely the behavior. I guess is there a reason only 1 should receive the gradients? I see that's what pytorch does at least. Not sure about tensorflow

**Comment by vikigenius** [comment]
I don't think there is a special reason, considering they also have an alternative amax function that propagates to 
multiple values. Although this naming and API is very confusing, and we should avoid this and decide on some consistent behavior.

https://pytorch.org/docs/stable/generated/torch.amax.html

**Comment by coreylowman** [comment]
Let's keep it as the behavior it is now for ease of implementation, and then we can add these details to the docstring (1. that it propagates values for all equal max values, and 2. that it is equivalent to torch.amax)
